telegram command: advanced status

flightcom opened the pull request:
This adds commands for Telegram:
- `/count` will allow to see number of active trades
- options added to existing `/status` command
  + `table` outputs trades information in a table format
  + `table short` outputs a shortened version

gcarq commented:
Hi, thanks for the PR

gcarq commented on freqtrade/rpc/telegram.py:
```diff
@@ -146,6 +157,71 @@ def _status(bot: Bot, update: Update) -> None:
+            trades_list.append(row)
+
+        columns = ['ID', 'Pair', 'Since', 'Amount', 'Open Rate',
+            'Close Rate', 'Cur. Rate', 'Close Profit', 'Profit',
```
Fix indentation. (see pylint)

gcarq commented on freqtrade/rpc/telegram.py:
```diff
@@ -146,6 +157,71 @@ def _status(bot: Bot, update: Update) -> None:
+
+        columns_short = ['Pair', 'Since', 'Profit']
+
+        if short_version == True:
```
this can be simplified to `if short_version:`

gcarq commented on freqtrade/rpc/telegram.py:
```diff
@@ -331,6 +407,29 @@ def _performance(bot: Bot, update: Update) -> None:
+
+    trades = Trade.query.filter(Trade.is_open.is_(True)).all()
+
+    current_trades_count = len(trades)
```
There is no need do declare these variables, just pass the values to format(). You can also use placeholders to make it more explicit what is being passed.

gcarq commented on freqtrade/rpc/telegram.py:
```diff
@@ -146,6 +157,71 @@ def _status(bot: Bot, update: Update) -> None:
+            row = [
+                trade.id,
+                trade.pair,
+                shorten_date(arrow.get(trade.open_date).humanize()),
```
I think the `shorten_date` function can be simplified if you pass `only_distance=True` to humanize()

gcarq commented on freqtrade/rpc/telegram.py:
```diff
@@ -146,6 +157,71 @@ def _status(bot: Bot, update: Update) -> None:
 
 
 @authorized_only
+def _status_table(bot: Bot, update: Update) -> None:
```
R:160, 0: Too many local variables (18/15) (too-many-locals) (See pylint)

gcarq commented:
@flightcom I really like the table short format.
I'm thinking about changing the default `/status` to `status table short` when this is merged, but I guess the width should reduced a little bit, because the last column is breaking (at least on the android client).

The normal table is not really helpful (at least for Telegram) because it is very hard to read on desktop and smartphone.

I left some comments, please also run pylint over the changed files.
A unittest would be also great, you can see how it is done in `tests/test_telegram.py` (not required for merging).

cheers

flightcom commented:
Hi @gcarq, thank you for your comments. I'm new in python and did not have pylint installed.
I changed the normal output to be the `short` one. Plus, I added unit tests for new commands.
Let me know if there is anything else that requires some changes.

Cheers!
